src/sherpa_ai/utils.py

```diff
@@ -325,3 +330,217 @@ def check_if_number_exist(result: str, source: str):
+    word2_set = set(word2)
+
+    lev_distance = edit_distance(word1, word2)
+    jaccard_sim = 1 - jaccard_distance(word1_set, word2_set)
+    long_len = max(len(word1), len(word2))
+    normalized_levenshtein = 1 - (lev_distance / long_len)
+
+    combined_metric = (levenshtein_constant * normalized_levenshtein) + (
+        (1 - levenshtein_constant) * jaccard_sim
+    )
+
+    return combined_metric
+
+def extract_entities(text): 
+    """
+        Extract entities of specific types 
+            NORP (Nationalities or Religious or Political Groups),
+            ORG (Organization),
+            GPE (Geopolitical Entity),
+            LOC (Location) using spaCy.
+        Args:
+        - text (str): Input text.
+
+        Returns:
+        List[str]: List of extracted entities.
+    """
+     
+    nlp = spacy.load("en_core_web_sm")
+    doc = nlp(text)
+    entity_types = ["NORP", "ORG", "GPE" , "LOC"]
+    filtered_entities = [ent.text for ent in doc.ents if ent.label_ in entity_types]
+
+    return filtered_entities
+
+
+def json_extractor(text):
+    """
+        Extract and parse JSON data from a text.
+
+        Args:
+        - text (str): Input text containing JSON data.
+
+        Returns:
+        dict: Parsed JSON data.
+    """
+
+    json_pattern = r"\{.*\}"
+    json_match = re.search(json_pattern, text)
+
+    if json_match:
+        json_data = json_match.group()
+        try:
+            parsed_json = json.loads(json_data)
+            return parsed_json
+        except json.JSONDecodeError as e:
+            return {}
+    else:
+        return {}
+
+
+def text_similarity_by_llm(
+    source_entity: List[str],
+    source,
+    result,
+    user_id=None,
+    team_id=None,
+):
+    """
+        Check if entities from a question are mentioned in some form inside the answer using a language model.
+
+        Args:
+        - source_entity (List[str]): List of entities from the question.
+        - source (str): Question text.
+        - result (str): Answer text.
+        - user_id (str): User ID (optional).
+        - team_id (str): Team ID (optional).
+
+        Returns:
+        dict: Result of the check containing 'entity_exist' and 'messages'.
+    """
+
+    llm = SherpaOpenAI(
+        temperature=cfg.TEMPRATURE,
+        openai_api_key=cfg.OPENAI_API_KEY,
+        user_id=user_id,
+        team_id=team_id,
+    )
+
+    instruction = f"""
+        I have a question and an answer. I want you to confirm if the entities from the question has been in some form inside the answer. 
+        Answer = {result}
+        Question = {source}
+        Entities inside the question = {source_entity}
+       """
+    instruction = (
+        instruction
+        + """
+           only return { entity_exist: 'true' , messages:"" } if entities are mentioned inside the answer in  
+           only return {entity_exist: 'false' , messages: " Entity x hasn't been mentioned inside the answer"} if the entity is not mentioned properly .
```

Comment:
only return {entity_exist: 'false' , messages: " Entity x is not mentioned within the answer"} if the entity is not mentioned within the answer .